torch/nn/functional.py

```diff
@@ -670,6 +670,10 @@ def binary_cross_entropy(input, target, weight=None, size_average=True):
                 sizeAverage is set to False, the losses are instead summed
                 for each minibatch.
     """
+    if not target.is_same_size(input):
+        warnings.warn("Using a target size ({}) that is different to the input size ({}) is deprecated. "
```

Comment:
@fmassa I believe the `BCELoss` module calls `BCECriterion.updateOutput` directly and not the function in `nn.functional` see here: https://github.com/pytorch/pytorch/blob/master/torch/nn/_functions/thnn/loss.py#L28  and here: https://github.com/pytorch/pytorch/blob/master/torch/nn/_functions/thnn/auto.py#L266